tools/specgen/specgen.go

```diff
@@ -70,20 +71,23 @@ var typesToDisambiguate = map[string]*CSModelType{
 			},
 		},
 	},
+	typeToKey(reflect.TypeOf(volume.Info{})):                 {Name: "VolumeInfo"},
+	typeToKey(reflect.TypeOf(volume.Topology{})):             {Name: "VolumeTopology"},
+	typeToKey(reflect.TypeOf(volume.Secret{})):               {Name: "VolumeSecret"},
```

Comment:
@galvesribeiro Is there a rule when a prefix should be used and when it should not? For example, some swarm types use a prefix while others do not. This situation is now similar to volume types. I have added the "Volume" prefix to these volume types to avoid clashes that would occur otherwise.

Reply:
Yeah, never had a rule before. But in the Swarm case, would make sense that all of them had the Swarm prefix. You are good with the Volume prefix in this case, thanks.